packages/integration-tests/src/test-e2e/sandbox/sandbox.test.template.ts

```diff
@@ -83,27 +84,33 @@ export const defineSandboxTest = (testProjectCreator: TestProjectCreator) => {
         void it(`[${testProjectCreator.name}] hot-swaps a change`, async () => {
           const updates = await testProject.getUpdates();
           if (updates.length > 0) {
-            const processController = ampxCli(
-              ['sandbox', '--dirToWatch', 'amplify'],
-              testProject.projectDirPath,
-              {
-                env: sharedSecretsEnv,
-              }
-            );
-
-            for (const update of updates) {
-              processController
-                .do(replaceFiles(update.replacements))
-                .do(waitForSandboxToBeginHotswappingResources());
-              if (update.deployThresholdSec) {
-                processController.do(
-                  ensureDeploymentTimeLessThan(update.deployThresholdSec)
+            // retry hotswapping resources if deployment time is higher than the threshold
+            await runWithRetry(
+              async () => {
+                const processController = ampxCli(
+                  ['sandbox', '--dirToWatch', 'amplify'],
+                  testProject.projectDirPath,
+                  {
+                    env: sharedSecretsEnv,
+                  }
```

Comment:
could you add a code comment explaining that it's important to keep initial deployment in retry loop ?

I.e. the app state must be reset in each retry for updates (hotswaps) to be non no-op.